### PR TITLE
Feature/add non default languages

### PR DESCRIPTION
**Pull request by royteeuwen**

### CODE REVIEW

**Comment by royteeuwen** [comment]
@kwin  go ahead and review :)

File: core/src/main/java/be/orbinson/aem/dictionarytranslator/utils/DictionaryUtil.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    }
+
+    /**
+     * Gets the language resource based on the jcr:language property
+     *
+     * @param dictionaryResource The dictionary resource
+     * @param language           The language
+     * @return the language resource if it exists
+     */
+    public static @Nullable Resource getLanguageResource(Resource dictionaryResource, String language) {
```
Comment (kwin) [inline core/src/main/java/be/orbinson/aem/dictionarytranslator/utils/DictionaryUtil.java:20]: I would rather put this into `DictionaryService` and clarify in the javadoc the difference to `getLanguages(Resource)`

Comment (kwin) [reply core/src/main/java/be/orbinson/aem/dictionarytranslator/utils/DictionaryUtil.java:20]: Maybe one can even combine both and let `getLanguages(Resource)` return a `Map<Locale,String>` (i.e. key = language and value = path/resource). Not sure how easy both are to pass to the servlets, though.

**Comment by royteeuwen** [comment]
@kwin  I did a major refactor, it now takes away the limitation of having the same node names and it also made everything more clear between which resources come from the resource provider and which are normal message entries (and no more labels ;))

A lot of the issues logged were also fixed while refactoring. You are welcome to test it out, I will make a release afterwards (unless you want a release on the current main branch)

File: ui.apps/src/main/content/jcr_root/apps/aem-dictionary-translator/content/granite/dialog/dictionary/key/delete/.content.xml
Comment (jmagdrod) [inline ui.apps/src/main/content/jcr_root/apps/aem-dictionary-translator/content/granite/dialog/dictionary/key/delete/.content.xml:1]: The key delete action is only removing those translations containing sling:message properties but not the entire key.
Is this the expected behavior? If so, I think that the dialog text should be renamed as "Are you sure you want to delete the translations for the selected keys?" and also I guess that a new key delete action must be added.

Steps to reproduce: 
1. Create a new key.
2. Add translations to some of the languages.
3. Delete key.
4. Behavior: translations are removed but the key is still present.

Comment (royteeuwen) [reply ui.apps/src/main/content/jcr_root/apps/aem-dictionary-translator/content/granite/dialog/dictionary/key/delete/.content.xml:1]: great catch, fixed ;)

**Comment by kwin** [comment]
Please go ahead. Only this minor thing appeared during our testing.

**Comment by royteeuwen** [comment]
@kwin  perfect, released! thanks for the thorough review :)

**Pull request merged**